test-ts/flows/end-to-end.test.ts

```diff
@@ -0,0 +1,258 @@
+      const [validator, validatorWallet] = await randomSigner(parseUnits("11000"));
+      validators.push(validator);
+      validatorAddresses.push(validators[i].address);
+
+      await registerValidatorGroup(groups[i]);
+      await registerValidator(groups[i], validators[i], validatorWallet);
+    }
+
+    accountsInstance = await hre.kit.contracts.getAccounts();
+    lockedGold = await hre.kit.contracts.getLockedGold();
+    election = await hre.kit.contracts.getElection();
+
+    const stakedCeloFactory: MockStakedCelo__factory = (
+      await hre.ethers.getContractFactory("MockStakedCelo")
+    ).connect(owner) as MockStakedCelo__factory;
+    stakedCelo = await stakedCeloFactory.deploy();
+  });
+
+  beforeEach(async () => {
+    await hre.deployments.fixture("TestAccount");
+    const owner = await hre.ethers.getNamedSigner("owner");
+    account = await hre.ethers.getContract("Account");
+    managerContract = await hre.ethers.getContract("Manager");
+    managerContract = managerContract.attach(await account.manager());
+    await account.connect(owner).setManager(managerContract.address);
+  });
+
+  it("deposit and withdraw", async () => {
+    const depositorInitialBalance = await depositor.getBalance();
+    console.log("depositorInitialBalance", depositorInitialBalance.toString());
+
+    managerContract.setDependencies(stakedCelo.address, account.address);
+
+    for (let i = 0; i < 3; i++) {
+      await managerContract.activateGroup(groupAddresses[i]);
+    }
+    await managerContract.connect(depositor).deposit({ value: 100 });
+    let stCelo = await stakedCelo.balanceOf(depositor.address);
+    expect(stCelo).to.eq(100);
+
+    const depositorAfterDepositBalance = await depositor.getBalance();
+    console.log("depositorAfterDepositBalance", depositorAfterDepositBalance.toString());
+
+    const groupList = await managerContract.getGroups();
+    console.log(`Groups: ${groupList}`);
+
+    await backendActivateAndVote(groupList);
+
+    const withdrawStakedCelo = await managerContract.connect(depositor).withdraw(100);
+    await withdrawStakedCelo.wait();
+
+    stCelo = await stakedCelo.balanceOf(depositor.address);
+    expect(stCelo).to.eq(0);
+
+    await backendWithdrawal(groupList);
+
+    const depositorBeforeWithdrawalBalance = await depositor.getBalance();
+    console.log("depositorBeforeWithdrawalBalance", depositorBeforeWithdrawalBalance.toString());
+
+    await timeTravel(LOCKED_GOLD_UNLOCKING_PERIOD);
+
+    const finishPendingWithdrawal = account.finishPendingWithdrawal(depositor.address, 0, 0);
+    await (await finishPendingWithdrawal).wait();
+    const depositorAfterWithdrawalBalance = await depositor.getBalance();
+    console.log("depositorAfterWithdrawalBalance", depositorAfterWithdrawalBalance.toString());
+    expect(depositorAfterWithdrawalBalance.gt(depositorBeforeWithdrawalBalance)).to.be.true;
+  });
+
+  async function backendActivateAndVote(groupList: string[]) {
```

Comment:
could we used  hardhat task here instead `hre.run("stakedCelo:account:activateAndVote")`  to avoid duplicating code?

Reply:
Absolutely ! Updated :-)